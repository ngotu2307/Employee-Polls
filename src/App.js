import logo from './logo.svg';
import './App.css';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import HomePage from "./components/HomePage";
import Login from './components/auth/Login';
import ErrorPage from './components/ErrorPage'
import Dashboard from './components/Dashboard';
import Leaderboard from './components/Leaderboard';
import PollPage from './components/PollPage';
import NewPollPage from './components/NewPollPage';
import Register from './components/auth/Register';
import RootLayout from './components/root/Root';

const router = createBrowserRouter([
  { 
    path: '/',
    element: <RootLayout />,
    children: [
      { index: true, element: <HomePage /> },
      { path: "error", element: <ErrorPage /> },
      { path: "register", element: <Register /> },
      { path: "dashboard", element: <Dashboard /> },
      { path: "leaderboard", element: <Leaderboard /> },
      { path: "poll/:id", element: <PollPage /> },
      { path: "newpoll", element: <NewPollPage /> }
    ]
  },

  { path: "/login", element: <Login /> }
])

function App() {
  return <RouterProvider router={router} />
}

export default App;
