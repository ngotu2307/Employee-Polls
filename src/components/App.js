import { Fragment, useEffect } from "react";
import { connect } from "react-redux";
import { handleInitialData } from "../actions/shared";
import Login from "./auth/Login";
import Register from "./auth/Register";
import Dashboard from "./dashboard/Dashboard";
import PollPage from "./PollPage";
import NewPollPage from "./NewPollPage";
import Leaderboard from "./leaderboard/Leaderboard";
import { Routes, Route } from "react-router-dom";
import LoadingBar from "react-redux-loading-bar";
import { Navigate } from "react-router-dom";
import NotFound from "./NotFound";

const App = (props) => {

  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  return (
    <Fragment>
      <LoadingBar />
      <Routes>
        <Route path="/" exact element={props.showLogin ? 	<Navigate to="/login" /> : <Navigate to="/dashboard" />} />
        <Route path="/login" exact element={<Login />} />
        <Route path="/register" exact element={<Register />} />
        <Route path="/dashboard" exact element={props.showLogin ? 	<Navigate to="/login" /> : <Dashboard />} />
        <Route path="/poll/:id" element={props.showLogin ? 	<Navigate to="/login" /> : <PollPage />} />
        <Route path="/newpoll" exact element={props.showLogin ? 	<Navigate to="/login" /> : <NewPollPage />} />
        <Route path="/leaderboard" exact element={props.showLogin ? 	<Navigate to="/login" /> : <Leaderboard />} />
        <Route path="/404" exact element={<NotFound />}/>
        <Route path="*" exact element={<NotFound />}/>
      </Routes>
    </Fragment>
  )
};

const mapStateToProps = ({ authedUser }) => ({
  showLogin: authedUser === null
});

export default connect(mapStateToProps)(App);
