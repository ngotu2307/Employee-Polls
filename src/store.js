import { createStore } from "redux";
import persistReducer from "./reducers";
import { persistStore } from 'redux-persist'
import middleware from "./middleware";

export const store = createStore(persistReducer, middleware);
export const persistor = persistStore(store)