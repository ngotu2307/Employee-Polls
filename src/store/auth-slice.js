import { createSlice, configureStore } from "@reduxjs/toolkit";
const initialAuthState = { isAuthen: false }

const authSlice = createSlice({
    name: 'auth',
    initialState: initialAuthState,
    reducers: {
        login(state) {
            state.isAuthen = true
        },
        logout(state) {
            state.isAuthen = false
        }
    }
})

export const authActions = authSlice.actions

export default authSlice.reducer;