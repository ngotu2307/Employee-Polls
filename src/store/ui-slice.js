import { createSlice, configureStore } from "@reduxjs/toolkit";
const initialUiState = { isAuthen: true }

const uiSlice = createSlice({
    name: 'auth',
    initialState: initialUiState,
    reducers: {
        login(state) {
            state.isAuthen = true
        },
        logout(state) {
            state.isAuthen = false
        }
    }
})

export const uiActions = uiSlice.actions

export default uiSlice.reducer;