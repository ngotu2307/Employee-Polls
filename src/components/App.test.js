import { render } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import { store, persistor } from "../store";
import { PersistGate } from 'redux-persist/integration/react'
import { BrowserRouter as Router } from "react-router-dom";

describe('App', () => {
  it("should render the component", () => {
    const component = render(
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <Router>
            <App />
          </Router>
        </PersistGate>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});