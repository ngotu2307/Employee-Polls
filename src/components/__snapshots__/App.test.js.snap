// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
      <div
        class="container mt-4"
      >
        <div
          class="row"
        >
          <div
            class="col-sm-12 d-flex flex-column justify-content-center"
          >
            <h2
              class="text-center"
            >
              Employee Polls
            </h2>
            <p />
            <div
              class="text-center"
            >
              <img
                class="rounded"
                src="employeepolls.jpeg"
                width="25%"
              />
            </div>
            <p />
            <h3
              class="text-center"
              data-testid="login-heading"
            >
              Log in
            </h3>
            <form>
              <div
                class="d-flex flex-column justify-content-center"
              >
                <div
                  class="form-group text-center"
                >
                  <label
                    for="userInput"
                  >
                    User
                  </label>
                  <input
                    class="form-control"
                    data-testid="username-input"
                    id="userInput"
                    placeholder="User"
                    type="text"
                  />
                </div>
                <br />
                <div
                  class="form-group text-center"
                >
                  <label
                    for="userPassword"
                  >
                    Password
                  </label>
                  <input
                    class="form-control"
                    data-testid="password-input"
                    id="userPassword"
                    placeholder="Password"
                    type="password"
                  />
                </div>
                <p />
                <div
                  class="text-center"
                >
                  <button
                    class="btn btn-primary"
                    data-testid="submit-button"
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </div>
            </form>
            <p
              class="text-danger"
              data-testid="error-message"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="direction: ltr;"
    >
      <div
        class=""
        style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
      />
      <div
        style="display: table; clear: both;"
      />
    </div>
    <div
      class="container mt-4"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-12 d-flex flex-column justify-content-center"
        >
          <h2
            class="text-center"
          >
            Employee Polls
          </h2>
          <p />
          <div
            class="text-center"
          >
            <img
              class="rounded"
              src="employeepolls.jpeg"
              width="25%"
            />
          </div>
          <p />
          <h3
            class="text-center"
            data-testid="login-heading"
          >
            Log in
          </h3>
          <form>
            <div
              class="d-flex flex-column justify-content-center"
            >
              <div
                class="form-group text-center"
              >
                <label
                  for="userInput"
                >
                  User
                </label>
                <input
                  class="form-control"
                  data-testid="username-input"
                  id="userInput"
                  placeholder="User"
                  type="text"
                />
              </div>
              <br />
              <div
                class="form-group text-center"
              >
                <label
                  for="userPassword"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  data-testid="password-input"
                  id="userPassword"
                  placeholder="Password"
                  type="password"
                />
              </div>
              <p />
              <div
                class="text-center"
              >
                <button
                  class="btn btn-primary"
                  data-testid="submit-button"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </div>
          </form>
          <p
            class="text-danger"
            data-testid="error-message"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
